hazelcast/src/main/java/com/hazelcast/nio/tcp/TcpIpConnectionManager.java

```diff
@@ -124,12 +125,14 @@ public TcpIpConnectionMonitor createNew(Address endpoint) {
     // to deal with this should only be used for the test lab. In the future we need to create a real fix to this problem,
     // but without this hack we can't do reliable benchmarking because the numbers have too much variation.
     private final boolean selectorImbalanceWorkaroundEnabled;
+    private HandlerBalancer handlerBalancer;
     private final Map<String, Integer> selectorIndexPerHostMap;
+    private final LoggingService loggingService;
 
-    public TcpIpConnectionManager(IOService ioService, ServerSocketChannel serverSocketChannel) {
+    public TcpIpConnectionManager(IOService ioService, ServerSocketChannel serverSocketChannel, LoggingService loggingService) {
         this.ioService = ioService;
         this.serverSocketChannel = serverSocketChannel;
-        this.logger = ioService.getLogger(TcpIpConnectionManager.class.getName());
+        this.logger = loggingService.getLogger(TcpIpConnectionManager.class.getName());
```

Comment:
yeah, I don't like it though. LoggingService should be the entry-point when it comes to obtaining loggers.